controller/favorite.controller.js

```diff
@@ -0,0 +1,38 @@
+const favoriteRepository = require("../repository/favoriteRepository");
+
+async function favorite(req, res) {
+    try{
+        const { params: { tutorId }, user } = req;
+
+        if (user.id === tutorId) {
+            return res.status(400).json({ message: "나에게는 찜할 수 없습니다." });
+        }
+
+        const selectRow = await favoriteRepository.selectTutorFavorite(user, tutorId);
+
+        if (selectRow === undefined) {
+            const insertRow = await favoriteRepository.insertTutorFavorite(user, tutorId);
+
+            if (insertRow.affectedRows === 1) {
+                return res.status(200).json({ message: "찜하기 완료" });
+            }
+        }
+
+        if (selectRow.user_id === user.id) {
+            const deleteRow = await favoriteRepository.deleteTutorFavorite(user, tutorId);
+
+            if (deleteRow.affectedRows === 1) {
+                return res.status(200).json({ message: "찜하기 취소 완료" });
+            }
+        } else {
+            return res.status(500).json({ message: "찜하기 취소 권한이 없습니다." });
+        }
```

Comment:
`selectRow.user_id`하고 `user.id`를 굳이 비교 할 필요가 있을까요????
검색 자체를 `user.id`로 하지 않나요??;;;

```suggestion
        let row;
        const result = {message:"찜하기 "}
        if (!selectRow) {
            row = await favoriteRepository.insertTutorFavorite(user, tutorId);
            result.message+="완료";
        }else{
            row = await favoriteRepository.deleteTutorFavorite(user, tutorId);
            result.message+="취소 완료";
        }
        
        if (row.affectedRows === 1) {
            return res.status(200).json(result);
        }
        return res.status(500).json({ message: "Invalid Server Error" });
```
이정도로 정리할 수 있을거같아요.